Libraries/oneMKL/matrix_mul/matrix_mul_mkl.cpp

```diff
@@ -0,0 +1,190 @@
+//==============================================================
+// Copyright � 2020 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <limits>
+#include "mkl.h"
+#include "mkl_blas_sycl.hpp"
+
+using namespace std;
+using namespace cl::sycl;
+
+// Matrix size constants
+#define SIZE 4800  // Must be a multiple of 8.
+#define M SIZE / 8
+#define N SIZE / 4
+#define P SIZE / 2
+
+/**
+ * Perform the matrix multiplication on host to verify results from mkl.
+ */
+int VerifyResult(double *c_back);
+
+int main() {
+  //
+  // Initialize data for Gemm
+  //
+  // C = alpha * op(A) * op(B)  + beta * C
+  //
+  mkl::transpose transA = mkl::transpose::nontrans;
+  mkl::transpose transB = mkl::transpose::nontrans;
+
+  // matrix data sizes
+  int m = M;
+  int n = P;
+  int k = N;
+
+  // leading dimensions of data
+  int ldA = m;
+  int ldB = k;
+  int ldC = m;
+
+  // set scalar fp values
+  double alpha = 1.0;
+  double beta = 0.0;
+
+  // 1D arrays on host side
+  double *A;
+  double *B;
+  double *C;
+
+  A = new double[M * N];
+  B = new double[N * P];
+  C = new double[M * P];
+
+  // prepare matrix data with column-major style
+  int i, j;
+  // A(M, N) is a matrix whose values are column number plus one
+  for (i = 0; i < N; i++)
+    for (j = 0; j < M; j++) A[i * M + j] = i + 1.0;
+
+  // B(N, P) is matrix whose values are row number plus one
+  for (i = 0; i < P; i++)
+    for (j = 0; j < N; j++) B[i * N + j] = j + 1.0;
+
+  cout << "Problem size: c(" << M << "," << P << ") = a(" << M << "," << N
+       << ") * b(" << N << "," << P << ")" << std::endl;
+
+  //
+  // Execute Gemm
+  //
+  auto asyncHandler = [&](cl::sycl::exception_list eL) {
+    for (auto &e : eL) {
+      try {
+        std::rethrow_exception(e);
+      } catch (cl::sycl::exception &e) {
+        std::cout << e.what() << std::endl;
+        std::cout << "fail" << std::endl;
```

Comment:
we should be using /n instead of std::endl

Reply:
> we should be using /n instead of std::endl

You mean `\n`? Ahem, anyway it should be `std::cerr`.

Reply:
Applied suggestion